DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,67 @@
+
```

Comment:
Can you write some intro about the SDK?
Ref: https://github.com/opensearch-project/OpenSearch/issues/1422

Reply:
sure, ill also include links to that issue and also the pluggable architecture blog post as well